### PR TITLE
[MPS] Add regression test for large inputs to `F.linear`

### PR DESCRIPTION
**Pull request by hvaara**
This PR adds a regression test for the issue reported in #122045. I was not able to reproduce on macOS > 13.

~Expect the first iteration of the tests to fail for macOS 13, but pass for 14 and 15.~

### CODE REVIEW

**Comment by hvaara** [comment]
I don't have easy access to macOS 13 right now. Can someone please approve running the tests and add the `ciflow/mps` label? 🙏

**Comment by hvaara** [comment]
Thanks to whoever approved the tests! ❤️ 

@pytorchbot label ciflow/mps

**Comment by hvaara** [comment]
@kulinseth @malfet Does this LGTY?

**Comment by hvaara** [comment]
@malfet thoughts on this?

**Comment by malfet** [comment]
Do you mind adding a fix that dispatches to the naive_mm on large input sizes? (or at least prints an error?)

File: test/test_mps.py [hunk 1]
```diff
@@ -2167,6 +2167,18 @@ def test_linear3D_no_bias(self):
     def test_linear3D_no_bias_backward(self):
         self._linear_helper(in_features=2, out_features=3, shape=((4, 5, 2)), bias=True, backward_pass=True)
 
+    @xfailIf(product_version < 14.0)
+    def test_linear_large(self):
+        x_cpu = torch.randn(9, 1024, 1, device='cpu')
```
Comment (malfet) [inline test/test_mps.py:2172]:
```suggestion
        # Regression test for https://github.com/pytorch/pytorch/issues/122045
        x_cpu = torch.randn(9, 1024, 1, device='cpu')
```

**Review by malfet — APPROVED** [review]
Though sure, it's better to have the test than not having one...

**Comment by hvaara** [comment]
Following the docs at https://github.com/pytorch/pytorch/wiki/Bot-commands.

I see some jobs that have been stuck for ~10h. 
[image](https://github.com/pytorch/pytorch/actions/runs/11045253136/job/30682558220?pr=136084) I don't know if these are still waiting in line, or the workflow has failed. Either way, something seems broken.

That means I can't run merge with the `-i` flag (although I might not be authorized to do that either). I'll try rebasing with `-r` and see what effect that has.

@pytorchbot merge -r

**Comment by hvaara** [comment]
Thanks @malfet! Just curious, in future events like these, what's the correct way for me to handle issues like these? Should I just escalate to someone with special powers (like yourself) instead of trying to sort it out myself?

**Comment by malfet** [comment]
> Just curious, in future events like these, what's the correct way for me to handle issues like these? Should I just escalate to someone with special powers (like yourself) instead of trying to sort it out myself?

I guess it depends on a case-by-case scenario. If you feel that failures are unrelated, feel free to try `merge -f`, but if unsure better to escalate.

**Comment by hvaara** [comment]
SGTM. I'm not authorized to issue `merge -f` commands, so I'll default to escalate with the suggested preferred command to run.

**Pull request closed without merging**